mage_ai/data_preparation/models/pipeline.py

```diff
@@ -499,18 +500,39 @@ async def to_dict_async(
 
     async def update(self, data, update_content=False):
         if 'name' in data and data['name'] != self.name:
+            from mage_ai.orchestration.db.models import Backfill, PipelineRun, PipelineSchedule
             """
             Rename pipeline folder
             """
+            old_uuid = self.uuid
             new_name = data['name']
             new_uuid = clean_name(new_name)
+
+            all_pipelines = self.get_all_pipelines(self.repo_path)
+            if new_uuid in all_pipelines:
+                raise Exception(f'Pipeline {new_uuid} already exists. Choose a different name.')
+
             old_pipeline_path = self.dir_path
             self.name = new_name
             self.uuid = new_uuid
             new_pipeline_path = self.dir_path
             os.rename(old_pipeline_path, new_pipeline_path)
             self.save()
 
+            # Migrate pipeline schedules
+            PipelineSchedule.query.filter(PipelineSchedule.pipeline_uuid == old_uuid).update({
+                PipelineSchedule.pipeline_uuid: new_uuid
+            }, synchronize_session=False)
+            # Migrate pipeline runs (block runs have foreign key ref to PipelineRun id)
+            PipelineRun.query.filter(PipelineRun.pipeline_uuid == old_uuid).update({
+                PipelineRun.pipeline_uuid: new_uuid
+            }, synchronize_session=False)
+            # Migrate backfills
+            Backfill.query.filter(Backfill.pipeline_uuid == old_uuid).update({
+                Backfill.pipeline_uuid: new_uuid
+            }, synchronize_session=False)
+            db_connection.session.commit()
```

Comment:
maybe create a separate method for these, and add the `@safe_db_query` decorator like here: https://github.com/mage-ai/mage-ai/blob/master/mage_ai/orchestration/db/models.py#L531